### DISCUSSION TITLE
How do you send a slash command?

### DISCUSSION THREAD

**Discussion by Notmoodo9** [rigwild/discord-self-bot-console#69] [Q&A] [answered]
This was answered in  [#31](https://github.com/rigwild/discord-self-bot-console/discussions/31) but no longer works, 
```
{
  id()

  const guildId = gid
  const channelId = cid

  // Search a command named `rank`, pick the first result
  const searchRes = await api.searchSlashCommand(channelId, 'help') // just a temp test 
  const command = searchRes.application_commands[0]
    console.log(command)
  // Get my rank, no command options
  await api.sendSlashCommand(guildId, channelId, command)
}
```

Command is always undefined even if i know for a fact that command should exist. Is there a way around this, I could even manually type in the command stuff but idk.

**Comment by Notmoodo9** [comment, marked as answer]
I fixed it kinda i changed how the commands are fetched, [changed code](https://github.com/Notmoodo9/discord-self-bot-console/blob/main/index.js). you can interact with it by doing this: 
```
const guildId = await api.getConfig().guildId;
const channelId = await api.getConfig().channelId; 
const cmd = "command_name"
const command = await api.searchAllSlashCommand(guildId, cmd)

await api.sendSlashCommand(guildId, channelId, command, (options) or nothing)   
```

note that every time searchAllSlashCommand is called it has to parse all the commands on the server so it is more efficient to store the command if you want to use it multiple times.

**Reply by Daxer1937** [reply]
Hi! I think I am really stupid or something, but the code is not working. I have used python before, but I don't use javascript, so i have been confused on how to use it (I know you need to first put in index.js, but when i try this code this code it just says (guildId, channelOrThreadId, command, commandOptions = []) => {
      const formData = new FormData()
      formData.append(
        'payload_json',
        JSON.stringify({
          type: 2,

**Reply by Notmoodo9** [reply]
> Hi! I think I am really stupid or something, but the code is not working. I have used python before, but I don't use javascript, so i have been confused on how to use it (I know you need to first put in index.js, but when i try this code this code it just says (guildId, channelOrThreadId, command, commandOptions = []) => {
> 
>       const formData = new FormData()
> 
>       formData.append(
> 
>         'payload_json',
> 
>         JSON.stringify({
> 
>           type: 2,
> 
> 

Is it possible for you to provide a screenshot? Of what it says.

**Reply by Daxer1937** [reply]
Okay!
![Screenshot 2024-08-14 1 07 20 PM](https://github.com/user-attachments/assets/ba662659-9189-4428-bba4-00a68eaaf6b5)

**Reply by Notmoodo9** [reply]
> Okay!
> 
> ![Screenshot 2024-08-14 1 07 20 PM](https://github.com/user-attachments/assets/ba662659-9189-4428-bba4-00a68eaaf6b5)
> 
> 
> 
> 

Now I'm a little confused lmao, I have a few questions.
1. What are you trying to do 
2. What browser are you using

I can provide example code once I know for sure 👍.

**Reply by Daxer1937** [reply]
I am trying to use a slash command in a discord server, but i was gonna change the code so it could be like it sends the command every certain period of time. I was gonna use the code the owner of this page has given, the Farm code, but it won't work. I am using Chrome.

**Reply by Daxer1937** [reply]
I think I'm just making it confusing,  and that im using the code wrong

**Reply by Notmoodo9** [reply]
> I am trying to use a slash command in a discord server, but i was gonna change the code so it could be like it sends the command every certain period of time. I was gonna use the code the owner of this page has given, the Farm code, but it won't work. I am using Chrome.

Ok so if you want it to work for slash commands use my index.js. https://github.com/Notmoodo9/discord-self-bot-console/blob/main/index.js its the same thing but slash commands should work so once you paste that into discord you can use this code:
```
;(async () => {
const guildId = await api.getConfig().guildId;
const channelId = await api.getConfig().channelId; 
const cmd = "command_name"
const command = await api.searchAllSlashCommand(guildId, cmd)
const delay = 1000; // 1 second

while(true) {
await api.sendSlashCommand(guildId, channelId, command, (options) or nothing) 
await api.delay(delay);
}
})()
```

FYI you would have to be on the guild window and channel, or you can hard code it by adding the channelID and guildID in. 
let me know if this works!

**Reply by Daxer1937** [reply]
I been getting this error
![image](https://github.com/user-attachments/assets/dfb98ec1-0f80-408e-9149-fe11a3ee291f)
Any solution?

**Reply by DeadLyBro** [reply]
@Daxer1937 
> I been getting this error
> [IMAGE]
> Any solution?

Check your code, there is `, (options) or nothing`. Change or remove that. 
I hope you fixed this before...
